picamera2/outputs/fileoutput.py

```diff
@@ -36,7 +37,8 @@ def outputframe(self, frame, keyframe=True):
             self._fileoutput.write(frame)
             self._fileoutput.flush()
 
-    def stop(self):
+    def stop(self, forcestop=False):
         """Close file handle and prevent recording"""
-        super().stop()
-        self._fileoutput.close()
+        if forcestop or not self._nostop:
+            super().stop()
+            self._fileoutput.close()
```

Comment:
I've not quite understood what `_nostop` and `forcestop` are for. Is is so that some outputs (when you have a list of several) can be made to stop by (for example) `stop_recording` but others will carry on?

Reply:
Yup that's right, its so others can carry on, for example the streaming file can carry on, whilst the circular buffer file output can be stopped

Reply:
So in the example I see that the socket output is created with `nostop=True`, but I couldn't see who was ever trying to stop it. Does that mean that the parameter wasn't actually necessary?

Reply:
Good point, I've just removed them, I think I had that in so that the encoder would stop all but ones with nostop, but we don't need that anymore because of how the example works.